test/sslapitest.c

```diff
@@ -9831,6 +9876,255 @@ static int test_load_dhfile(void)
 #endif
 }
 
+#if !defined(OSSL_NO_USEABLE_TLS1_3) && !defined(OPENSSL_NO_COMP) && (defined(ZLIB) || !defined(OPENSSL_NO_BROTLI) || !defined(OPENSSL_NO_ZSTD))
+/*
+ * Test 0 = Skip compression
+ * Test 1 = libssl does the compression
+ * Test 2 = error out during compression
+ * Test 3 = libssl does compression, client authentication
+ * Test 4 = app does the compression
+ */
+/* This callback just says to skip compression */
+static int cert_comp_cb0(SSL *ssl, uint16_t alg, void* arg)
+{
+    *(int*)arg = alg;
+    return 0;
+}
+/* This callback just says to do the compression */
+static int cert_comp_cb1(SSL *ssl, uint16_t alg, void* arg)
+{
+    *(int*)arg = alg;
+    return alg;
+}
+/* This callback just says to error out */
+static int cert_comp_cb2(SSL *ssl, uint16_t alg, void* arg)
+{
+    return -1;
+}
+static int cert_comp_cb4(SSL *ssl, uint16_t alg, void* arg)
```

Comment:
It would be nice to have a comment explaining its purpose like the other callbacks above